core/algorithms/onpolicy_sync/engine.py

```diff
@@ -906,13 +911,24 @@ def update(self, rollouts: RolloutStorage):
         #     )
         # )
 
-    def make_offpolicy_iterator(self, data_iterator_builder: Callable[..., Iterator]):
+    def make_offpolicy_iterator(
+        self, data_iterator_builder: Callable[..., Iterator],
+    ):
         stage = self.training_pipeline.current_stage
-        kwargs = stage.offpolicy_component.data_iterator_kwargs_generator()
-        assert (
-            "current_worker" not in kwargs
-        ), "current_worker is reserved for the engine to be passed to the offpolicy data iterator builder"
-        return data_iterator_builder(current_worker=self.worker_id, **kwargs)
+
+        if self.num_workers == 1:
+            rollouts_per_worker = [self.num_samplers]
+        else:
+            rollouts_per_worker = self.num_samplers_per_worker
+
+        # common seed for all workers (in case we wish to shuffle the full dataset before iterating on one partition)
+        seed = self.advance_seed(self.seed, call_worker_seeds=False)
```

Comment:
This is a great idea.

Reply:
Yes. It's not straightforward to use (we need to move around the RNG state), but I think it can be useful.